### ISSUE TITLE
Compilation error with ProGuard enabled due to FindBugs annotation `SuppressFBWarnings`

### ISSUE THREAD

**Issue by Whathecode** [objectbox/objectbox-java#1011]
**Describe the bug**
Using ProGuard 7.2.0-beta2 and ObjectBox 3.0.1, I get the following ProGuard warning during compilation:

> Warning: io.objectbox.ideasonly.ModelModifier$PropertyModifier: can't find referenced class edu.umd.cs.findbugs.annotations.SuppressFBWarnings

Seemingly the `SuppressFBWarnings` class on `ModelModifier.PropertyModifier` can't be found. Since this seems part of a static analysis for [FindBugs](http://findbugs.sourceforge.net/), I can simply disable the warning using the following proguard rules:

> -dontwarn io.objectbox.ideasonly.ModelModifier$PropertyModifier

**Basic info (please complete the following information):**
 - ObjectBox version: 3.0.1
 - Reproducibility: always

**Expected behavior**
For this to work out-of-the-box. Perhaps by including the `dontwarn` rule [in the packaged `objectbox-java.pro`](https://github.com/objectbox/objectbox-java/blob/main/objectbox-java/src/main/resources/META-INF/proguard/objectbox-java.pro).

**Additional context**
I am currently upgrading from ProGuard which is integrated as part of the Android Gradle Plugin 7.1.0-beta1 to the separate Proguard Gradle Plugin in order to see whether the latest version resolves a potential obfuscation bug I am running into. Therefore, I have currently enabled ProGuard on my **debug build**. Not certain whether this impacts ObjectBox, but figured this might be relevant to mention. Regardless, I just also tested in **release**, and the same warning shows up.

**Comment by greenrobot-team** [comment]
Thanks. Your title says compilation error, but the message indicates it is only a warning. Are you sure that warning is the cause of the error? Or does the app actually compile?

Anyhow, yes, we should probably add a dontwarn rule for the SpotBugs annotations.

**Comment by greenrobot-team** [comment]
@Whathecode Just to make clear, this warning only happens with a separate ProGuard plugin (7.2.0-beta2)? I failed to repro with Android Gradle Plugin 7.1.0-beta1.

As the affected class isn't used anywhere this might be a ProGuard regression...

**Comment by greenrobot-team** [comment]
Next update will add this rule:
```
-dontwarn edu.umd.cs.findbugs.annotations
```

@Whathecode Can you verify if this one also works?

**Comment by Whathecode** [comment]
I can check, but I doubt it. Afaik the `-dontwarn` rule needs to specify the class from where the warning originates. It doesn't filter out warnings for _missing_ classes.

**Comment by Whathecode** [comment]
On top of that, `edu.ud.cs.findbugs.annotations` is an incomplete [class filter](https://www.guardsquare.com/manual/configuration/usage). If what is intended is all classes in that namespace you need to append `.*`, or `.**` to include deeper namespaces. Note that the only class that triggered a warning for me was the one I listed before.

As expected (stated in the previous message), the rule you suggest does not resolve the warning.

**Comment by greenrobot-team** [comment]
You are right, mixed up syntax.
So
```
-dontwarn edu.umd.cs.findbugs.annotations.**
```
does not work as well? I vaguely remember adding a rule for a referenced class worked as well.

**Comment by Whathecode** [comment]
Just checked, and that indeed works!

Interesting. That is not clearly documented in the ProGuard documentation.

**Comment by greenrobot-team** [comment]
Thanks for verifying! Next update will (actually) add the more specific rule:
```
-dontwarn edu.umd.cs.findbugs.annotations.SuppressFBWarnings
```

**Comment by greenrobot-team** [comment]
This is included in the `3.1.0` release.
